src/custom-apps/components/WebhookEvents/WebhookEvents.tsx

```diff
@@ -1,97 +1,270 @@
+                  </ListItemCell>
+                </ListItem>
+              </ListHeader>
+              {object &&
+                EventTypes[tab][object] &&
+                EventTypes[tab][object].map((event, idx) => (
+                  <ListItem className={classes.listItem} key={idx}>
+                    <ListItemCell className={classes.listItemCell}>
+                      <strong>
+                        {capitalize(event.toLowerCase().replaceAll("_", " "))}
+                      </strong>
+                    </ListItemCell>
+                    <ListItemCell>
+                      <Checkbox
+                        name={`${tab}Events`}
+                        checked={data[`${tab}Events`].includes(
+                          getEventName(object, event),
+                        )}
+                        value={getEventName(object, event)}
+                        onChange={handleEventChange}
+                        className={classes.checkbox}
+                      />
+                    </ListItemCell>
+                  </ListItem>
+                ))}
+            </List>
+          </div>
+        </Grid>
+      </Card>
+    </>
   );
 };
 WebhookEvents.displayName = "WebhookEvents";
 export default WebhookEvents;
+
+type Actions = string[];
+
+const AsyncWebhookTypes: Record<string, Actions> = {
+  ADDRESS: ["CREATED", "UPDATED", "DELETED"],
+  APP: ["INSTALLED", "UPDATED", "DELETED"],
+  ATTRIBUTE: ["CREATED", "UPDATED", "DELETED"],
+  CATEGORY: ["CREATED", "UPDATED", "DELETED"],
+  CHANNEL: ["CREATED", "UPDATED", "DELETED"],
+  GIFT_CARD: ["CREATED", "UPDATED", "DELETED", "STATUS_CHANGED"],
+
+  CHECKOUT: ["CREATED", "UPDATED", "DELETED"],
+  COLLECTION: ["CREATED", "UPDATED", "DELETED"],
+  CUSTOMER: ["CREATED", "UPDATED", "DELETED"],
+  FULFILLMENT: ["CREATED"],
+  INVOICE: ["DELETED", "REQUESTED", "SENT"],
+  MENU: ["CREATED", "UPDATED", "DELETED"],
+  ORDER: [
+    "CANCELLED",
+    "CONFIRMED",
+    "CREATED",
+    "FULFILLED",
+    "FULLY_PAID",
+    "UPDATED",
+  ],
+  PAGE: ["CREATED", "UPDATED", "DELETED"],
+  PRODUCT: ["CREATED", "UPDATED", "DELETED"],
+  PRODUCT_VARIANT: ["CREATED", "UPDATED", "DELETED"],
+  SALE: ["CREATED", "UPDATED", "DELETED", "TOGGLE"],
+  SHIPPING_PRICE: ["CREATED", "UPDATED", "DELETED"],
+  SHIPPING_ZONE: ["CREATED", "UPDATED", "DELETED"],
+  STAFF: ["CREATED", "UPDATED", "DELETED"],
+  TRANSLATION: ["ACTION_REQUEST", "CREATED", "UPDATED"],
+  VOUCHER: ["CREATED", "UPDATED", "DELETED"],
+  WAREHOUSE: ["CREATED", "UPDATED", "DELETED"],
+};
+
+const SyncWebhookTypes: Record<string, Actions> = {
+  PAYMENT: [
+    "AUTHORIZE",
+    "CAPTURE",
+    "CONFIRM",
+    "LIST_GATEWAYS",
+    "PROCESS",
+    "REFUND",
+    "VOID",
+  ],
+  CHECKOUT: ["CALCULATE_TAXES", "FILTER_SHIPPING_METHODS"],
+  ORDER: ["CALCULATE_TAXES", "FILTER_SHIPPING_METHODS"],
+  SHIPPING: ["LIST_METHODS_FOR_CHECKOUT"],
+};
+
+const EventTypes = {
+  async: AsyncWebhookTypes,
+  sync: SyncWebhookTypes,
+};
+
```

Comment:
**issue**: Previously, updating webhook enums required only building types from schema. I believe that typing it this way will result in having to edit this manually which is more work for developers and bug-prone